.circleci/soltest_all.sh

```diff
@@ -28,10 +28,46 @@ set -e
 
 REPODIR="$(realpath $(dirname $0)/..)"
 
-EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 ${REPODIR}/.circleci/soltest.sh
+# This number has to equal the number of steps below.
+STEPS=11
```

Comment:
You could avoid hard-coding the number of steps by using arrays and the `${#...}` syntax for getting array length:
```suggestion
EVM_VALUES=(homestead byzantium constantinople petersburg istanbul)
OPTIMIZE_VALUES=(0 1)
STEPS=$(( 1 + ${#EVM_VALUES[@]} * ${#OPTIMIZE_VALUES[@]} ))
```

And then below:
```bash
for OPTIMIZE in ${OPTIMIZE_VALUES[@]}; do
    for EVM in ${EVM_VALUES[@]}; do
```